src/tls13.c

```diff
@@ -4086,7 +4086,7 @@ int DoTls13ServerHello(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
     }
 
 #ifdef WOLFSSL_DTLS_CID
-    if (ssl->options.useDtlsCID)
+    if (ssl->options.useDtlsCID && *extMsgType == server_hello)
```

Comment:
This should have already been checked in `TLSX_Parse`. Is this getting past that check somehow?

Reply:
@rizlik any feedback on Juliusz’s comment?

Reply:
The extension is not parsed in the `HRR`, indeed `DtlsCIDOnExtensionsParsed` should not be invoked during `HRR` parsing, the check is ok. Here we are calling `DtlsCIDOnExtensionsParsed` that's related but different from parsing.   `DtlsCIDOnExtensionsParsed` is invoked after parsing the server hello extensions, but gets invoked on an HRR, this was a bug. 
`DtlsCIDOnExtensionsParsed` can't be invoked by `CID_PARSE` as it needed to be invoked also when the `CID` is not present (and so `CID_PARSE` isn't invoked) to detect that CID wasn't negotiated by the other peer.